opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -5,10 +5,15 @@
 from opentelemetry.trace.status import Status
 from opentelemetry.util import types
 
+MAX_NUM_ATTRIBUTES = 32
+
 
 class Span(abc.ABC):
     """A span represents a single operation within a trace."""
 
+    def __init__(self, attributes: types.Attributes = None) -> None:
```

Comment:
I think this class is meant to be like an interface, so you wouldn't want a concrete implementation of a constructor in here.

I don't think we need any interface for the constructor, because users of this interface won't need to worry about constructing it